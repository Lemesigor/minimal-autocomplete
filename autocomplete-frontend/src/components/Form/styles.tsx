import { css } from '@emotion/react';

export const formStyles = css`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 3.5rem; /* This corresponds to mt-14 in Tailwind */
`;

