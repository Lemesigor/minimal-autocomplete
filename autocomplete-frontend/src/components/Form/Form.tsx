/** @jsxImportSource @emotion/react */
import {formStyles} from "./styles";

export const Form  = () => {
    return (
        <section css={formStyles}>
            <input type={"text"} placeholder={"Search"} />
        </section>
    )
}