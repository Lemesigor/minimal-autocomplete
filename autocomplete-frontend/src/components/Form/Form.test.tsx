import {render} from "@testing-library/react";

import {Form} from "./Form";

describe("Form", () => {
    it("should render the form", () => {
        const {container} = render(<Form />);
        expect(container).toMatchSnapshot();
    });

    it("should have input field", () => {
        const {getByTestId} = render(<Form />);
        expect(getByTestId("autocomplete-input")).toBeInTheDocument();
    })
});