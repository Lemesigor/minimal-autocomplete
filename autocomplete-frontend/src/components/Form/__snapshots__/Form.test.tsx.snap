// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render the form 1`] = `
<div>
  <section
    class="css-1ms9z5j"
  >
    <div
      class="css-11aqn9s"
    >
      <input
        class="css-6asy4p"
        data-testid="autocomplete-input"
        placeholder="Search an English Word"
        type="text"
        value=""
      />
    </div>
  </section>
</div>
`;
